### PR TITLE
FlatProtonLogKsiLogTGun

### PR DESCRIPTION
**Pull request by grzanka**

### CODE REVIEW

File: IOMC/ParticleGuns/interface/FlatProtonLogKsiLogTGun.h [hunk 1]
```diff
@@ -0,0 +1,75 @@
+#ifndef IOMC_FlatProtonLogKsiLogTGun_FlatProtonLogKsiLogTGun_h
+#define IOMC_FlatProtonLogKsiLogTGun_FlatProtonLogKsiLogTGun_h
```
Comment (grzanka) [inline IOMC/ParticleGuns/interface/FlatProtonLogKsiLogTGun.h]: Change the header accordingly to package directory, i.e. 
`IOMC_FlatProtonLogKsiLogTGun_FlatProtonLogKsiLogTGun_h` -> `IOMC_ParticleGuns_FlatProtonLogKsiLogTGun_h`

File: IOMC/ParticleGuns/plugins/BuildFile.xml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<library   name="IOMCFlatProtonLogKsiLogTGunPlugins" file="*.cc">
```
Comment (grzanka) [inline IOMC/ParticleGuns/plugins/BuildFile.xml]: Change the library name to follow the directory names.

File: IOMC/ParticleGuns/plugins/FlatProtonLogKsiLogTGun.cc [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+  max_iter_number_ = pset.getUntrackedParameter<int>("MaximumIterationNumber");
+  
+  log_ksi_distribution_ = pset.getUntrackedParameter<bool>("LogKsiDistShape");
+  log_t_distribution_ = pset.getUntrackedParameter<bool>("LogTDistShape");
+  z_symetric_mode_ = pset.getUntrackedParameter<bool>("GenerateMirroredProtons");
+  instanceLabel = pset.getUntrackedParameter<std::string>("instanceLabel");
+
+  produces<edm::HepMCProduct>();
+
+//  std::cout << "Internal FlatRandomEGun is initialized" << std::endl ;
```
Comment (grzanka) [inline IOMC/ParticleGuns/plugins/FlatProtonLogKsiLogTGun.cc]: Is it needed ?

File: IOMC/ParticleGuns/plugins/FlatProtonLogKsiLogTGun.cc [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+
+
+void FlatProtonLogKsiLogTGun::produce(edm::Event& e, const edm::EventSetup& es)
+{
+  if(verbosity_)
+    std::cout<<" FlatProtonLogKsiLogTGun : Begin New Event Generation"<<std::endl;
+
+
+  edm::Service<edm::RandomNumberGenerator> rng;
+  //long seed = (long)(rng->mySeed());
```
Comment (grzanka) [inline IOMC/ParticleGuns/plugins/FlatProtonLogKsiLogTGun.cc]: remove

File: Configuration/Generator/python/Beta90_cfi.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import FWCore.ParameterSet.Config as cms
```
Comment (grzanka) [inline Configuration/Generator/python/Beta90_cfi.py]: Remove this file

File: Configuration/Generator/python/TestEnergy6500GeV_cfi.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
+import FWCore.ParameterSet.Config as cms
```
Comment (grzanka) [inline Configuration/Generator/python/TestEnergy6500GeV_cfi.py]: Change name of this file to:
FlatLogKsiLogT_pp_13TeV_cfi.py

File: Configuration/Generator/python/TestEnergy6500GeV_cfi.py [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+generator = cms.EDProducer("FlatProtonLogKsiLogTGun",
+    LogKsiDistShape = cms.untracked.bool(True),
+    LogTDistShape = cms.untracked.bool(True),
+    GenerateMirroredProtons = cms.untracked.bool(False),
+
+    RightArm = cms.untracked.bool(True),
+    LeftArm = cms.untracked.bool(True),
+    NominalEnergy = cms.untracked.double(6500.0), # [GeV]
```
Comment (grzanka) [inline Configuration/Generator/python/TestEnergy6500GeV_cfi.py]: Add comment that:
 - center of mass energy is 13 TeV and nominal energy is 6.5 TeV

File: IOMC/ParticleGuns/plugins/BuildFile.xml [hunk 8]
```diff
@@ -0,0 +1,10 @@
+<library   name="IOMCParticleGunsFlatProtonLogKsiLogTGunPlugins" file="*.cc">
```
Comment (grzanka) [inline IOMC/ParticleGuns/plugins/BuildFile.xml]: Is this file needed at all ? It seems the `plugins` directory doesn't exists here: https://github.com/cms-sw/cmssw/tree/CMSSW_8_1_X/IOMC/ParticleGuns and other guns seems to work.

File: IOMC/ParticleGuns/interface/FlatProtonLogKsiLogTGun.h [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+#include <memory>
+#include "boost/shared_ptr.hpp"
+
+using namespace edm;
+namespace HepMC {
+	class GenEvent;
+}
+
+
+class FlatProtonLogKsiLogTGun : public edm::EDProducer
```
Comment (grzanka) [inline IOMC/ParticleGuns/interface/FlatProtonLogKsiLogTGun.h]: Seems to be very different from typical gun implementation, see i.e. inheritance here:
https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_X/IOMC/ParticleGuns/interface/BaseFlatGunProducer.h

Check also new CTPPS gun, submitted to official CMSSW repo:
https://github.com/cms-sw/cmssw/pull/15096/files

File: IOMC/ParticleGuns/plugins/BuildFile.xml (hunk 8 above)
Comment (ziaber) [inline IOMC/ParticleGuns/plugins/BuildFile.xml]: Seems to work without it.
